src/compiler/scanner.ts

```diff
@@ -320,13 +320,20 @@ namespace ts {
 
     /* @internal */
     export function computeLineAndCharacterOfPosition(lineStarts: number[], position: number) {
+        Debug.assert(lineStarts.length && lineStarts[0] === 0, "the first line should start at position 0");

```

Comment:
can we remove these asserts from here. the server calls this function at least once for every call. we are not removing asserts in retail version and this code has not changed in a relay long time. so i do not see the value.